app_dart/test/request_handlers/luci_status_test.dart

```diff
@@ -234,6 +237,55 @@ void main() {
     );
   });
 
+  test('Reschedules an infra failure', () async {
```

Comment:
Does this also check that we do not get into a reschedule loop?

Reply:
Added another test to verify the negative side of what we do when we have exceeded the retries.

This test does verify that we increment the retries counter on the user_data.  The new test verifies that if that value >= the config value, we set the status to failure.